gretl/src/main/java/ch/so/agi/gretl/tasks/impl/Ili2pgAbstractTask.java

```diff
@@ -8,427 +8,213 @@
-
-        this.database = new Connector(databaseUri, databaseUser, databasePassword);
-    }
-
-    public void setDbschema(String dbschema) {
-        this.dbschema = dbschema;
-    }
-
-    public void setProxy(String proxy) {
-        this.proxy = proxy;
-    }
-
-    public void setProxyPort(Integer proxyPort) {
-        this.proxyPort = proxyPort;
-    }
-
-    public void setModeldir(String modeldir) {
-        this.modeldir = modeldir;
-    }
-
-    public void setModels(String models) {
-        this.models = models;
-    }
-
-    public void setDataset(Object dataset) {
-        this.dataset = dataset;
-    }
-
-    public void setBaskets(String baskets) {
-        this.baskets = baskets;
-    }
-
-    public void setTopics(String topics) {
-        this.topics = topics;
-    }
-
-    public void setImportTid(Boolean importTid) {
-        this.importTid = importTid;
-    }
-
-    public void setExportTid(Boolean exportTid) {
-        this.exportTid = exportTid;
-    }
-
-    public void setImportBid(Boolean importBid) {
-        this.importBid = importBid;
-    }
-
-    public void setPreScript(File preScript) {
-        this.preScript = preScript;
-    }
-
-    public void setPostScript(File postScript) {
-        this.postScript = postScript;
-    }
-
-    public void setDeleteData(Boolean deleteData) {
-        this.deleteData = deleteData;
-    }
-
-    public void setLogFile(Object logFile) {
-        this.logFile = logFile;
-    }
-
-    public void setTrace(Boolean trace) {
-        this.trace = trace;
-    }
-
-    public void setValidConfigFile(File validConfigFile) {
-        this.validConfigFile = validConfigFile;
-    }
-
-    public void setDisableValidation(Boolean disableValidation) {
-        this.disableValidation = disableValidation;
-    }
-
-    public void setDisableAreaValidation(Boolean disableAreaValidation) {
-        this.disableAreaValidation = disableAreaValidation;
-    }
-
-    public void setForceTypeValidation(Boolean forceTypeValidation) {
-        this.forceTypeValidation = forceTypeValidation;
-    }
-
-    public void setStrokeArcs(Boolean strokeArcs) {
-        this.strokeArcs = strokeArcs;
-    }
-
-    public void setSkipPolygonBuilding(Boolean skipPolygonBuilding) {
-        this.skipPolygonBuilding = skipPolygonBuilding;
-    }
-
-    public void setSkipGeometryErrors(Boolean skipGeometryErrors) {
-        this.skipGeometryErrors = skipGeometryErrors;
-    }
-
-    public void setIligml20(Boolean iligml20) {
-        this.iligml20 = iligml20;
-    }
-
-    public void setDisableRounding(Boolean disableRounding) {
-        this.disableRounding = disableRounding;
-    }
-
-    public void setFailOnException(Boolean failOnException) {
-        this.failOnException = failOnException;
-    }
-
-    public void setDatasetSubstring(Range<Integer> datasetSubstring) {
-        this.datasetSubstring = datasetSubstring;
-    }
+    public abstract ListProperty<Integer> getDatasetSubstring();
```

Comment:
Hier musste ich von Range<Integer> auf ListProperty<Integer> wechseln, da ListProperty/Property Range als Typ nicht akzeptieren. Sprich ich musste die Logik etwas anpassen um dieselbe Funktionalität "nachzubilden". Für den Enduser hat das in seinem build.gradle File nur zur Folge, dass er die Range nun in eine Liste wandeln muss (z.b. (0...4).toList(). Mehr ändert sich nicht.

Man könnte hier auch zwei Properties verwenden (from/to). Ich fände das ein bisschen klarer als die aktuelle Lösung. Wir können das aber ja noch besprechen oder vielleicht hast du noch einen weiteren Vorschlag?

Reply:
@edigonzales

Reply:
@Saela Wir lassen es fürs Erste so.